File: server/src/users/models/user.model.ts
import {
  Table,
  Model,
  Column,
  DataType,
  BelongsToMany,
  HasOne,
  HasMany,
} from 'sequelize-typescript';
import { Expert } from 'src/experts/models/expert.model';
import { Order } from 'src/orders/models/order.model';
import { Role } from 'src/roles/models/role.model';
import { UserRole } from './user-role.model';

interface UserCreationAttrs {
  last_name?: string;
  name: string;
  second_name?: string;
  phone: string;
  birth_date?: Date;
  email: string;
  password: string;
}

@Table({ tableName: 'users' })
export class User extends Model<User, UserCreationAttrs> {
  @Column({
    type: DataType.INTEGER,
    unique: true,
    autoIncrement: true,
    primaryKey: true,
  })
  id: number;

  @Column({ type: DataType.STRING, unique: false, allowNull: true })
  last_name: string;

  @Column({ type: DataType.STRING, unique: false, allowNull: false })
  name: string;

  @Column({ type: DataType.STRING, unique: false, allowNull: true })
  second_name: string;

  @Column({ type: DataType.STRING, unique: true, allowNull: false })
  phone: string;

  @Column({ type: DataType.DATE, unique: false, allowNull: true })
  birth_date: Date;

  @Column({ type: DataType.STRING, unique: true, allowNull: false })
  email: string;

  @Column({ type: DataType.STRING, unique: false, allowNull: false })
  password: string;

  @HasMany(() => Order)
  orders: Order[];

  @BelongsToMany(() => Role, () => UserRole)
  roles: Role[];

  @HasOne(() => Expert, { onDelete: 'CASCADE' })
  expert: Expert;
}

File: server/modules/orm/models/user.model.go
package model

type User struct {
	ID         uint64   `gorm:"primary_key, AUTO_INCREMENT"`
	LastName   string   `json:"last_name"`
	Name       string   `json:"name"`
	SecondName string   `json:"second_name"`
	BirthDate  string   `json:"birth_date"`
	Email      string   `json:"email"`
	Password   string   `json:"password"`
	Phone      string   `json:"phone"`
	Role       string   `json:"role"`
	Position   string   `json:"position"`
	ImageUrl   string   `json:"image_url"`
	Cert       string   `json:"cert"`
	Directions string   `json:"directions"`
	Misc       string   `json:"misc"`
	Works      []*Order `binding:"required" gorm:"foreignkey:ExpertID; constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
	Orders     []*Order `binding:"required" gorm:"foreignkey:ClientID; constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
}
